File: backend/app/Models/Withdrawal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Withdrawal extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'amount',
        'payment_method',
        'account_number',
        'account_name',
        'status',
        'reference_id',
        'admin_note',
        'completed_at',
    ];

    protected $casts = [
        'amount' => 'decimal:2',
        'completed_at' => 'datetime',
    ];

    /**
     * Get the user that owns the withdrawal.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Boot function logic for generating reference_id
     */
    protected static function boot()
    {
        parent::boot();

        static::creating(function ($withdrawal) {
            if (!$withdrawal->reference_id) {
                $withdrawal->reference_id = 'WD-' . strtoupper(\Illuminate\Support\Str::random(8));
            }
        });
    }
}

File: enstore/lib/core/models/withdrawal.dart
class Withdrawal {
  final int id;
  final String referenceId;
  final double amount;
  final String paymentMethod;
  final String accountNumber;
  final String accountName;
  final String status;
  final String createdAt;

  Withdrawal({
    required this.id,
    required this.referenceId,
    required this.amount,
    required this.paymentMethod,
    required this.accountNumber,
    required this.accountName,
    required this.status,
    required this.createdAt,
  });

  factory Withdrawal.fromJson(Map<String, dynamic> json) {
    return Withdrawal(
      id: json['id'] ?? 0,
      referenceId: json['reference_id'] ?? '',
      amount: double.tryParse(json['amount']?.toString() ?? '0') ?? 0,
      paymentMethod: json['payment_method'] ?? '',
      accountNumber: json['account_number'] ?? '',
      accountName: json['account_name'] ?? '',
      status: json['status'] ?? 'pending',
      createdAt: json['created_at'] ?? '',
    );
  }
}
